### PR TITLE
feat(Automation details): Service automation details modal

### PR DESCRIPTION
**Pull request by kmunene**
**What does this PR do?**

- Adds the service automation details modal
- Adds SCSS for the service automation modal
- Adds tests.

**Description of task to be completed?**
A user should see the details of a service automation in a modal.

**How should this be manually tested?**

- Clone and set up the project using the instruction in the README.
- $ git checkout to `ft-modal-to-show-service-automation-details` branch which contains the changes.
- Use $ npm start to start the application
- Visit http://localhost:3000/.
- On the report page, click the information (i) icons to view the details of the various service automation.

**What are the relevant pivotal tracker stories?**
[[#160437219]](https://www.pivotaltracker.com/story/show/160437219) - Sub-task

**Screenshots:**
##### Slack automation details
screen shot 2018-11-14 at 15 43 05

##### Email automation details
screen shot 2018-11-14 at 15 43 28

##### Freckle automation details
screen shot 2018-11-09 at 12 19 57

### CODE REVIEW

**Comment by IAMOTZ** [comment]
I don't think there is a need to show the status icon in any of the modal again:
screen shot 2018-11-13 at 12 01 50 pm

**Comment by IAMOTZ** [comment]
On the Email Autionmation Modal:
- we need to see the partner name and not partner tag.
- we need to see the to-email(who the email was sent to) and not the devloper's email
- we plan to send multiple emails for either the onboarding or offboarding phase, that is yet to be reflected on the modal

**Comment by IAMOTZ** [comment]
On the Slack Automation Modal:
- Show the status for each slack channel addition/removal
- Show the partner name on the slack automation modal
- Add an `#` behined each channel name to make it easier to identify

File: src/components/ReportPage/index.jsx [hunk 1]
```diff
@@ -222,16 +234,28 @@ class ReportPage extends PureComponent {
...
+      this.setState({ modalContent: report, modalType: type });
+    }
+    if (type === 'email') {
+      this.setState({ modalContent: report, modalType: type });
+    }
+    if (type === 'freckle') {
+      this.setState({ modalContent: report, modalType: type });
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (kmunene) [reply src/components/ReportPage/index.jsx]: Sure, will refactor that.

**Pull request merged**